rp/src/key.rs

```diff
@@ -0,0 +1,105 @@
+use std::{fs, path::Path};
+
+use anyhow::{anyhow, Result};
+use wireguard_keys::Privkey;
+
+use rosenpass::protocol::{SPk, SSk};
+use rosenpass_cipher_traits::Kem;
+use rosenpass_ciphers::kem::StaticKem;
+use rosenpass_secret_memory::file::StoreSecret as _;
+
+pub fn genkey(private_keys_dir: &Path) -> Result<()> {
+    if private_keys_dir.exists() {
+        return Err(anyhow!("Directory {:?} already exists", private_keys_dir));
+    }
+
+    fs::create_dir_all(private_keys_dir)?;
```

Comment:
For private keys it is extremely important to enforce proper permissions. Please ensure that the directory is created with `0700` permissions and files in it with `0600`. If it already exists, check if the permissions match, otherwise panic.

Reply:
Resolved in dce6d5d6d929e1ae3487843fde62829b8cc0ca24.